torch/_refs/__init__.py

```diff
@@ -1655,48 +1655,108 @@ def item(a: TensorLikeType) -> NumberType:
     return number_type(prims.item(a))
 
 
-# TODO: extend to device and layout similar to
-# aten/src/ATen/native/TensorConversions.cpp:to_will_alias
 def _to_will_alias(
     a: TensorLikeType,
-    dtype: torch.dtype,
-    copy: bool,
-    memory_format: torch.memory_format,
+    device: Optional[torch.device],
+    dtype: Optional[torch.dtype],
+    copy: Optional[bool],
+    memory_format: Optional[torch.memory_format],
+    layout: Optional[torch.layout],
 ) -> bool:
     return (
         not copy
-        and a.dtype == dtype
+        and (device is None or a.device == device)
+        and (dtype is None or a.dtype == dtype)
+        and (layout is None or a.layout == layout)
         and (
-            memory_format == torch.preserve_format
-            or a.is_contiguous(memory_format=memory_format)
+            memory_format is None
+            or memory_format == torch.preserve_format
+            or utils.is_contiguous_for_memory_format(a, memory_format=memory_format)
         )
     )
 
 
-# TODO: there are multiple overloads of to, but we only support one here
-# aten/src/ATen/native/native_functions.yaml lists all overloads
-# Missing overloads:
-# to.dtype_layout
-# to.device
-# to.other
-# https://github.com/pytorch/pytorch/issues/84264
+def _to_dispatch(
+    device=None,
+    dtype=None,
+    non_blocking=None,
+    copy=None,
+    memory_format=None,
+    layout=None,
+    pin_memory=None,
```

Comment:
layout and pin_memory are not modified in this function, maybe we shouldn't pass it?

Reply:
I see `layout` is used below, but it does seem like `pin_memory` isn't (because we don't have any prims that support it?)

Reply:
Instead of removing unused params from the signature, let's throw an unsupported error when they're passed